api.py

```diff
@@ -44,82 +74,171 @@ async def status():
+    logger.debug("Attempting to validate schema from JSON data...")
     if data:
-        if isinstance(data, str):
-            try:
-                json_to_validate = json.loads(data)
-            except JSONDecodeError:
-                logger.info("Could not parse JSON", status=400)
-                return Response(status_code=400, content="Could not parse JSON")
-        elif isinstance(data, dict):
+        # Sets `json_to_validate` to the data received if it is a dictionary as data does not require parsing
+        if isinstance(data, dict):
+            logger.info("JSON data received as dictionary - parsing not required")
             json_to_validate = data
+        # Sets `json_to_validate` to the parsed data if it is a string
+        elif isinstance(data, str):
+            json_to_validate = parse_json(data)
+        # Returns an error response if the data received is not a string or dictionary
+        else:
+            logger.error(
+                "Invalid data type received for validation (expected string or dictionary)",
+                data_type=type(data),
+                status=400,
+            )
+            return Response(
+                status_code=400, content="Invalid data type received for validation"
+            )
+    else:
+        logger.error("No JSON data provided for validation", status=400)
+        return Response(status_code=400, content="No JSON data provided for validation")
 
     response = {}
     try:
+        logger.debug(
+            "Sending JSON data to AJV Schema Validator service...",
+            url=AJV_VALIDATOR_URL,
+        )
+        # Posts JSON data to AJV Validator service and returns a response containing any errors
         ajv_response = requests.post(
             AJV_VALIDATOR_URL, json=json_to_validate, timeout=10
         )
+        # Returns errors in the response if AJV Validator service returned any errors
         if ajv_response_dict := ajv_response.json():
             response["errors"] = ajv_response_dict["errors"]
-            logger.info("Schema validator returned errors", status=400)
+            logger.warning(
+                "AJV Schema Validator service returned errors",
+                status=400,
+                errors=response["errors"],
+            )
             return response, 400
 
     except RequestException:
-        logger.info("AJV Schema validator service unavailable")
-        return json.dumps(obj={}, error="AJV Schema validator service unavailable")
+        logger.error("AJV Schema Validator service unavailable")
+        return json.dumps(obj={}, error="AJV Schema Validator service unavailable")
+
+    logger.info("AJV Schema Validator service returned no errors", status=200)
 
     validator = QuestionnaireValidator(json_to_validate)
+    logger.debug(
+        "Attempting to validate questionnaire schema contents with Questionnaire Validator...",
+        questionnaire_title=json_to_validate.get("title"),
```

Comment:
One last request, can we extract `survey_id` and `form_type` here as well? Title can be slightly misleading sometimes (and repeated for multiple schemas) while with `title` combined with `survey_id`, `form_type` there are no two identical surveys? The only thing is `form_type` is not required field but appears in all business schemas.

Reply:
Ideally, we would like schema file name here but that would require a lot of extra work..